### PR TITLE
Allow `ds` to take multiple values to repeat for a count

### LINKED ISSUE #722: [Feature request] Variable arguments for `ds` like `db`/`dw`/`dl`
[RASM](http://www.roudoudou.com/rasm/rasm_v112_en.pdf), like rgbasm, allows `ds N, V` to act like `db V, V, V, ...` with `N` values. However, it goes further and lets the `ds` take an arbitrary number of run-lengths. So `ds 3, "A", 4, "B"` would be like `db "AAABBBB"`.

Since the other data directives `db`, `dw`, `dl` all take multiple parameters, I think it would make sense for `ds` as well.

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #722

### CODE REVIEW

**Comment by ISSOtm** [comment]
We might actually be hitting edge cases here... what would you expect `ds 10, @` to produce ? `ds 10, @, @ + 1`?

**Comment by Rangi42** [comment]
I would expect `@` to be eagerly evaluated. So this:

```
SECTION "test", ROM0[3]
  ds 6, @
```

would output `00 00 00 03 03 03 03 03 03`, which it does. Likewise, my expectation of this:

```
SECTION "test", ROM0[3]
  ds 7, @, @ + 1
```

would be to output `00 00 00 03 04 03 04 03 04 03`, which it does.

**Comment by Rangi42** [comment]
If you do want lazy evaluation, it can be done with a macro:

```
__ds_lazy_item: MACRO
	rept 1 + (\1 % _NARG)
		shift
	endr
	db \1
ENDM

ds_lazy: MACRO
n = \1
	shift
	for i, n
		__ds_lazy_item i, \#
	endr
ENDM

SECTION "test", ROM0
	ds_lazy 10, @ ; 00 01 02 03 04 05 06 07 08 09
	ds_lazy 10, @, @ + 1 ; 00 02 02 04 04 06 06 08 08 0a
```

File: test/asm/ds-byte.asm [hunk 1]
```diff
@@ -6,3 +6,4 @@ Label:
 	ds 5, .other - Label - 5 ; Expressions should work...
 	ds 60, .last - Label     ; ...even if not constant
 .last
+	ds 11, $67, $89
```
Comment (ISSOtm) [inline test/asm/ds-byte.asm:9]: Please add a test for `ds N, @` in both fixed and floating sections. RGBLINK's evaluation of PC may yield surprising results.

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: test/asm/ds-byte.asm (hunk 1 above)
Comment (Rangi42) [reply test/asm/ds-byte.asm:9]: Ah, I had forgotten about link-time evaluation. You're right, it's different at asm vs link time, even for single `ds 10, @`.

Comment (ISSOtm) [reply test/asm/ds-byte.asm:9]: \> Mfw we find out a bug we've had for a couple versions

Comment (Rangi42) [reply test/asm/ds-byte.asm:9]: Arguably not a bug, or at least, a bug that I don't see an alternative to. It's like how `SIN(someLabel)` is different at link-time than at asm-time if `someLabel` is floating.

Comment (ISSOtm) [reply test/asm/ds-byte.asm:9]: Could be fixed with a new patch type.

Comment (Rangi42) [reply test/asm/ds-byte.asm:9]: Can that be a separate issue+PR from this one?

Comment (ISSOtm) [reply test/asm/ds-byte.asm:9]: Should be, and I'd prefer to merge it before this one, so that this one can add the relevant tests and pass them.

**Comment by Rangi42** [comment]
Rebased after #737 merged.

File: src/asm/section.c [hunk 2]
```diff
@@ -636,20 +636,24 @@ void out_RelByte(struct Expression *expr, uint32_t pcShift)
...
  * it is an absolute value in disguise.
  */
-void out_RelBytes(struct Expression *expr, uint32_t n)
+void out_RelBytes(uint32_t n, struct Expression *exprs, size_t size)
 {
 	checkcodesection();
 	reserveSpace(n);
 
-	for (uint32_t i = 0; i < n; i++) {
+	for (size_t i = 0; i < n; i++) {
```
Comment (ISSOtm) [inline src/asm/section.c:644]: `uint32_t` is better, because it's the type of `n`.

**Review by ISSOtm — CHANGES_REQUESTED** [review]
